IssetOnPropertyObjectToPropertyExistsRector > Call to a member function getType() on null

Scanning this file:

```php
<?php declare(strict_types=1);

namespace App\Services\Cloud\Provider\Onedrive\Client\Files\Traits;

use Collator;

trait Sort
{
    /**
     * @param array $contents
     *
     * @return array
     */
    protected function sort(array $contents): array
    {
        $collator = new Collator('es');

        usort($contents, fn ($a, $b) => $this->sortItem($collator, $a, $b));

        return $contents;
    }

    /**
     * @param \Collator $collator
     * @param object $a
     * @param object $b
     *
     * @return int
     */
    protected function sortItem(Collator $collator, object $a, object $b): int
    {
        $aFolder = isset($a->folder);
        $bFolder = isset($b->folder);

        if ($aFolder && $bFolder) {
            return $collator->compare($a->name, $b->name);
        }

        if ($aFolder) {
            return -1;
        }

        if ($bFolder) {
            return 1;
        }

        return $collator->compare($a->name, $b->name);
    }
}
```

I get:

```
 [ERROR] Could not process "app/Services/Cloud/Provider/Onedrive/Client/Files/Traits/Sort.php" file by                  
         "Rector\CodeQuality\Rector\Isset_\IssetOnPropertyObjectToPropertyExistsRector", due to:                        
         "Call to a member function getType() on null".         
```

Thanks!

File: https://github.com/rectorphp/rector/blob/48eaa5c950e2aff3c7442d2b6b13794be7dfde66/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php#L93

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -87,8 +87,13 @@ CODE_SAMPLE
             /** @var Expr $object */
             $object = $issetVar->var->getAttribute(AttributeKey::ORIGINAL_NODE);
 
-            /** @var Scope $scope */
+            /** @var Scope $scope|null */
             $scope = $object->getAttribute(AttributeKey::SCOPE);
+
+            if (is_null($scope)) {
+                continue;
+            }
+
             /** @var ThisType|ObjectType $type */
             $type = $scope->getType($object);
 

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -91,6 +91,10 @@ CODE_SAMPLE
             /** @var ThisType|ObjectType $type */
             $type = $this->getType($object);
 
+            if ($type === null) {
+                continue;
+            }
+
             if ($type instanceof ThisType) {
                 $newNodes[] = new NotIdentical($issetVar, $this->createNull());
                 continue;

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -88,7 +88,7 @@ CODE_SAMPLE
             /** @var Expr $object */
             $object = $issetVar->var->getAttribute(AttributeKey::ORIGINAL_NODE);
 
-            /** @var ThisType|ObjectType $type */
+            /** @var ThisType|ObjectType|null $type */
             $type = $this->getType($object);
 
             if ($type === null) {

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -87,7 +87,7 @@ CODE_SAMPLE
             /** @var Expr $object */
             $object = $issetVar->var->getAttribute(AttributeKey::ORIGINAL_NODE);
 
-            /** @var Scope $scope|null */
+            /** @var Scope|null $scope */
             $scope = $object->getAttribute(AttributeKey::SCOPE);
 
             if ($scope === null) {

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -90,7 +90,7 @@ CODE_SAMPLE
             /** @var Scope $scope|null */
             $scope = $object->getAttribute(AttributeKey::SCOPE);
 
-            if (is_null($scope)) {
+            if (! $scope) {
                 continue;
             }
 

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -90,7 +90,7 @@ CODE_SAMPLE
             /** @var Scope $scope|null */
             $scope = $object->getAttribute(AttributeKey::SCOPE);
 
-            if (! $scope) {
+            if ($scope === null) {
                 continue;
             }
 

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -19,6 +19,7 @@ use PhpParser\Node\Scalar\String_;
 use PHPStan\Analyser\Scope;
 use PHPStan\Type\ObjectType;
 use PHPStan\Type\ThisType;
+use PHPStan\Type\Type;
 use Rector\Core\Rector\AbstractRector;
 use Rector\NodeTypeResolver\Node\AttributeKey;
 use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
@@ -87,15 +88,8 @@ CODE_SAMPLE
             /** @var Expr $object */
             $object = $issetVar->var->getAttribute(AttributeKey::ORIGINAL_NODE);
 
-            /** @var Scope|null $scope */
-            $scope = $object->getAttribute(AttributeKey::SCOPE);
-
-            if ($scope === null) {
-                continue;
-            }
-
             /** @var ThisType|ObjectType $type */
-            $type = $scope->getType($object);
+            $type = $this->getType($object);
 
             if ($type instanceof ThisType) {
                 $newNodes[] = new NotIdentical($issetVar, $this->createNull());
@@ -124,6 +118,18 @@ CODE_SAMPLE
         }
 
         return $this->createReturnNodes($newNodes);
+    }
+
+    private function getType(Expr $expr): ?Type
+    {
+        /** @var Scope|null $scope */
+        $scope = $expr->getAttribute(AttributeKey::SCOPE);
+
+        if ($scope === null) {
+            return null;
+        }
+
+        return $scope->getType($expr);
     }
 
     private function replaceToPropertyExistsWithNullCheck(

--- a/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
+++ b/rules/code-quality/src/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector.php
@@ -90,19 +90,15 @@ CODE_SAMPLE
 
             /** @var ThisType|ObjectType|null $type */
             $type = $this->getType($object);
+            /** @var Identifier|Variable $name */
+            $name = $issetVar->name;
 
-            if ($type === null) {
+            if ($type === null || ! $name instanceof Identifier) {
                 continue;
             }
 
             if ($type instanceof ThisType) {
                 $newNodes[] = new NotIdentical($issetVar, $this->createNull());
-                continue;
-            }
-
-            /** @var Identifier|Variable $name */
-            $name = $issetVar->name;
-            if (! $name instanceof Identifier) {
                 continue;
             }
 

--- a/rules/code-quality/tests/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector/Fixture/skip_null_type.php.inc
+++ b/rules/code-quality/tests/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector/Fixture/skip_null_type.php.inc
@@ -1,0 +1,35 @@
+<?php
+
+namespace Rector\CodeQuality\Tests\Rector\Isset_\IssetOnPropertyObjectToPropertyExistsRector\Fixture;
+
+trait SkipNullType
+{
+    protected function sort(array $contents): array
+    {
+        $collator = new Collator('es');
+
+        usort($contents, fn ($a, $b) => $this->sortItem($collator, $a, $b));
+
+        return $contents;
+    }
+
+    protected function sortItem(Collator $collator, object $a, object $b): int
+    {
+        $aFolder = isset($a->folder);
+        $bFolder = isset($b->folder);
+
+        if ($aFolder && $bFolder) {
+            return $collator->compare($a->name, $b->name);
+        }
+
+        if ($aFolder) {
+            return -1;
+        }
+
+        if ($bFolder) {
+            return 1;
+        }
+
+        return $collator->compare($a->name, $b->name);
+    }
+}

--- a/rules/code-quality/tests/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector/Fixture/skip_null_type.php.inc
+++ b/rules/code-quality/tests/Rector/Isset_/IssetOnPropertyObjectToPropertyExistsRector/Fixture/skip_null_type.php.inc
@@ -8,7 +8,7 @@ trait SkipNullType
     {
         $collator = new Collator('es');
 
-        usort($contents, fn ($a, $b) => $this->sortItem($collator, $a, $b));
+        usort($contents, function ($a, $b) { return $this->sortItem($collator, $a, $b); });
 
         return $contents;
     }

